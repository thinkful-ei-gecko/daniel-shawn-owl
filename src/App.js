import React from 'react';
import ChatEvents from './ChatEvents';
import ParticipantsList from './ParticipantsList';



function App(props) {
    return(<main className="App">
        <header className="App-header">
            <h1>OWL</h1>
        </header>
        <div className="Owl-app">
            <ParticipantsList Participants={props.Participants}></ParticipantsList>
        </div> 

    </main>);
}

export default App;
