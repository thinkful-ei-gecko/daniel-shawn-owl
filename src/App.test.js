import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import renderer from 'react-test-renderer'
import Participants from './Participants'
import ChatEvents from './ChatEvents'


it('Renders without crashing', () =>{
    const div = document.createElement('div');
    ReactDOM.render(<App Participants = {Participants} ChatEvents = {ChatEvents}/>, div);
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<App Participants = {Participants} ChatEvents = {ChatEvents} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});