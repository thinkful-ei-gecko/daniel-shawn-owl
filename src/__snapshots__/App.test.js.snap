// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      OWL
    </h1>
  </header>
  <div
    className="Owl-app"
  >
    <button
      className="chat-toggle"
      onClick={[Function]}
    >
      Chat
    </button>
    <button
      className="Participant-toggle"
      onClick={[Function]}
    >
      Participants
    </button>
    <div
      className="ParticipantView"
    >
      <div
        className="ParticipantList"
      >
        <div
          className="Participant"
        >
          <div>
            <img
              className="Avatar"
              src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
            />
          </div>
          <ul
            className="NameOnStageGroup"
          >
            <li
              className="Name"
            >
              Koren Templeton
            </li>
            <li
              className="OnStage"
            >
              <span>
                on stage - 
              </span>
              <a
                href="#"
              >
                Boot
              </a>
            </li>
            <li>
              Koren Templeton left
            </li>
          </ul>
        </div>
        <div
          className="Participant"
        >
          <div>
            <img
              className="Avatar"
              src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
            />
          </div>
          <ul
            className="NameOnStageGroup"
          >
            <li
              className="Name"
            >
              Caty Flucker
            </li>
            <li
              className="OnStage"
            >
              <span>
                on stage - 
              </span>
              <a
                href="#"
              >
                Boot
              </a>
            </li>
            <li>
              Caty Flucker left
            </li>
          </ul>
        </div>
        <div
          className="Participant"
        >
          <div>
            <img
              className="Avatar"
              src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
            />
          </div>
          <ul
            className="NameOnStageGroup"
          >
            <li
              className="Name"
            >
              Frank Runciman
            </li>
            <li
              className="OnStage"
            >
              <span>
                on stage - 
              </span>
              <a
                href="#"
              >
                Boot
              </a>
            </li>
            <li>
              Frank Runciman left
            </li>
          </ul>
        </div>
        <div
          className="Participant"
        >
          <div>
            <img
              className="Avatar"
              src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
            />
          </div>
          <ul
            className="NameOnStageGroup"
          >
            <li
              className="Name"
            >
              Ashla Attwool
            </li>
            <li
              className="OnStage"
            >
              <span>
                on stage - 
              </span>
              <a
                href="#"
              >
                Boot
              </a>
            </li>
            <li>
              Ashla Attwool left
            </li>
          </ul>
        </div>
        <div
          className="Participant"
        >
          <div>
            <img
              className="Avatar"
              src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
            />
          </div>
          <ul
            className="NameOnStageGroup"
          >
            <li
              className="Name"
            >
              Axe Kubicka
            </li>
            <li
              className="OnStage"
            >
              <span>
                on stage - 
              </span>
              <a
                href="#"
              >
                Boot
              </a>
            </li>
            <li>
              Axe Kubicka left
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="chatlist"
    >
      <div
        className="ChatWindow"
      >
        <ul
          className="chats"
        >
          <li>
            <img
              className="Avatar"
              src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
            />
            <span>
              Axe Kubicka
            </span>
            <span>
              7
              :
              48
            </span>
            <p>
              Hello world
            </p>
          </li>
          <li>
            Caty Flucker
             
            thumbs-up
          </li>
          <li>
            Frank Runciman
             
            thumbs-down
          </li>
          <li>
            Koren Templeton
             
            raise-hand
          </li>
          <li>
            Koren Templeton
             
            clap
          </li>
          <li>
            Koren Templeton
             
            join
          </li>
          <li>
            Ashla Attwool
             
            leave
          </li>
          <li>
            Axe Kubicka
             
            join-stage
          </li>
          <li>
            Axe Kubicka
             
            leave-stage
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="stage"
  >
    <div
      className="onStage"
    >
      <div
        className="stageIcon"
      >
        <img
          alt="Koren Templeton"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
        <div
          className="StageName"
        >
          <p>
            Koren Templeton
          </p>
        </div>
      </div>
      <div
        className="stageIcon"
      >
        <img
          alt="Ashla Attwool"
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
        <div
          className="StageName"
        >
          <p>
            Ashla Attwool
          </p>
        </div>
      </div>
    </div>
  </div>
</main>
`;
